Fixed infinite loop in _modman_get_module()

poebel opened the pull request:
$root is the real path with de-referenced symlinks while "$(pwd)" may contain symlinks
Reverted previous workaround

colinmollenhour commented:
Are you sure that 'pwd -P' is going to be supported by all major platforms? E.g. Mac OS and GNU

poebel commented:
Not totally, but if it's not supported it should already fail on line 48.
Actually it might be a better idea to remove the "-P" on said line instead, will test this tomorrow.

poebel commented:
As said earlier, I've updated the fix to remove all usages of "pwd -P" and added an additional check if we've reached the root level of the filesystem to prevent any further loops.

colinmollenhour commented:
Thanks!
